src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -551,52 +524,24 @@ private static void setClientProxyParams(String host, HttpClient client) {
         }
     }
 
-    private static int executeMethod(HttpClient client, HttpMethod method) throws IOException {
-        Jenkins jenkins = Jenkins.getInstance();
-        ProxyConfiguration proxyConfig = null;
-        if (jenkins != null) {
-            proxyConfig = jenkins.proxy;
-        }
-
-        Proxy proxy = Proxy.NO_PROXY;
-        if (proxyConfig != null) {
-            proxy = proxyConfig.createProxy(getMethodHost(method));
-        }
-
-        if (proxy.type() != Proxy.Type.DIRECT) {
-            final InetSocketAddress proxyAddress = (InetSocketAddress)proxy.address();
-            LOGGER.fine("Jenkins proxy: " + proxy.address());
-
-            final HostConfiguration hc = new HostConfiguration(client.getHostConfiguration());
-            hc.setProxy(proxyAddress.getHostString(), proxyAddress.getPort());
-
-            String username = proxyConfig.getUserName();
-            String password = proxyConfig.getPassword();
-            if (username != null && !"".equals(username.trim())) {
-                LOGGER.fine("Using proxy authentication (user=" + username + ")");
-
-                final HttpState state = new HttpState();
-                state.setProxyCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
-                return client.executeMethod(hc, method, state);
-            } else {
-                return client.executeMethod(hc, method);
+    private int executeMethod(HttpMethod httpMethod) throws InterruptedException, IOException {
+        int status = client.executeMethod(httpMethod);
+        while (status == API_RATE_LIMIT_CODE) {
+            if (Thread.interrupted()) {
+                throw new InterruptedException();
             }
-        } else {
-            return client.executeMethod(method);
+            LOGGER.fine("Bitbucket Cloud API rate limit reached, sleeping for 5 sec then retry...");
```

Comment:
leave a `// TODO` comment to the effect that once Bitbucket start reporting the rate limit reset timestamp, we should instead sleep until that timestamp rather than just blindly waiting 5 seconds

Reply:
done